src/adsp_config.c

```diff
@@ -374,7 +399,9 @@ static int parse_adsp(const toml_table_t *toml, struct parse_ctx *pctx, struct a
 		/* non-configurable fields */
 
 		/* configurable fields */
-		parse_str_key(mem_zone, &ctx, "type", zone_name, sizeof(zone_name), &ret);
+		zone_name[sizeof(zone_name) - 1] = 0;
+		parse_str_key(mem_zone, &ctx, "type", (uint8_t *)zone_name,
+			      sizeof(zone_name) - 1, &ret);
```

Comment:
it seems reasonable to move clearing last string-array byte to `parse_str_key()` function... or even `memset` to zero what will guarantee deterministic output when such a mechanism is not implemented on enclosing structures.

Reply:
> The destination is simply not a string.

It's not a string but it needs null termination here... so it's quite slippery. Setting null termination manually seems little ugly for me. Maybe a cleanest solution will be to just rename this parsing function to something like `parse_printable_key` (version without adding a null termination) and return length of a string then create a simple wrapper function:

```
static void parse_str_key(const toml_table_t *table, struct parse_ctx *ctx, const char *key,
			  char *dst, int capacity, int *error)
{
    int len = parse_printable_key(table, ctx, key, dst, capacity - 1, error);
    dst[len] = 0;
}
```

It's just and idea.

Reply:
Done something close to what @ktrzcinx suggested please have a look.

Also added the too long string to the "too long" error message.